phpBB/phpbb/extension/extension_interface.php

```diff
@@ -27,6 +27,13 @@
 	public function is_enableable();
 
 	/**
+	* Indicate why the extension can not be enabled.
+	*
+	* @return string|array with actual processed language string(s) (not language keys), or empty
+	*/
+	public function not_enableable_reason();
```

Comment:
It seems for me that both points can be satisfied as well as BC retained: if we inject new class property (say, ```$errs = []```), and then just assign it a value(s) inside ```is_enableable()``` in case of ```false```. The ```is_enableable()``` will still return bool, while in case of ```false``` we could get error mesagge with some class method like ```get_errors()``` returning the ```$errs``` propetry value.